### PR TITLE
Update getGEMatrix to used cached esets when combining multiple matrices

### PR DESCRIPTION
**Pull request by helenmiller16**
When multiple matrix names are passed into `ISCon$getGEMatrix` they do not use cached expression sets. Creating expression sets tends to take the most time in `getGEMatrix`. Here's my attempt at changing it to use cached esets when available. I understand that there may be some side effects that I've missed and am happy to chat and/or refactor if needed.

### CODE REVIEW

File: tests/testthat/test-getParticipantData.R [hunk 1]
```diff
@@ -28,16 +47,38 @@ test_getParticipantData("hla_typing")
...
   expect_is(EM, "ExpressionSet")
-  expect_is(EM, "ExpressionSet")
-  expect_gt(ncol(Biobase::exprs(EM)), 0)
   expect_gt(nrow(Biobase::exprs(EM)), 0)
+
+  ids <- CONNECTIONS$ALL$.__enclos_env__$private$.getParticipantIdsFromGroup("gem_test")
+  expect_lte(length(unique(EM$participant_id)), length(ids))
+
   # In summary, no gene is NA
   expect_false(any(is.na(fData(EM)$gene_symbol)))
```
Comment (ehfhcrc) [inline tests/testthat/test-getParticipantData.R]: Would you mind calling the `fData` function explicitly with `Biobase::fData`? This way it can be run as a single file easily without remembering to load the package.

File: tests/testthat/test-getGEMatrix.R [hunk 2]
```diff
@@ -69,13 +70,36 @@ test_that("get_multiple matrices summary without cache error", {
...
-    reload = TRUE
+                        outputType = "summary",
+                        annotation = "latest",
+                        reload = TRUE
   )
   test_EM(EM, summary = TRUE)
 })
 
+test_that("get multiple matrices summary from different studies", {
+  EM <- sdy$getGEMatrix(c("SDY269_PBMC_TIV_Geo", "SDY180_WholeBlood_Grp2Pneunomax23_Geo"), outputType = "summary", annotation = "latest")
```
Comment (ehfhcrc) [inline tests/testthat/test-getGEMatrix.R]: Since these have longer names here, can you please stack the arguments like the example above (e.g. "get multiple matrices summary with reload)? Makes it easier to follow.

File: tests/testthat/test-getGEMatrix.R [hunk 3]
```diff
@@ -69,13 +70,36 @@ test_that("get_multiple matrices summary without cache error", {
...
+  test_EM(EM, summary = TRUE)
+})
+
+test_that("loading from cache works correctly", {
+  # Should load both matrices from cache
+  expect_message(sdy$getGEMatrix(c("SDY269_PBMC_TIV_Geo", "SDY180_WholeBlood_Grp2Pneunomax23_Geo"), outputType = "summary", annotation = "latest"),
+               "(cache)|(Combining ExpressionSets)", all = TRUE)
+  # summary with a different annotation should download a new matrix
```
Comment (ehfhcrc) [inline tests/testthat/test-getGEMatrix.R]: Let's add a newline before each new `expect_message` to make things more readable.

**Comment by ehfhcrc** [comment]
Retroactively posting checklist notes:
https://www.notion.so/rglab/Review-pidGetGEMx-PR-properly-7a0286b095e944ee8414be2cd50f8613

TL;DR:
- Functional requirements met (allow for single eset from group of participants and handles reloading correctly)
- New tests accurately capture older functionality that was not properly tested as well as new functionality
- Separate branch created to add related modifications for best practices (use of `fread` instead of `read.table` and `vapply` instead of `sapply` as well as handler for NULL matrixName arg)
- Future work should include cross-cohort normalization and batch effect correction for cross-study participant group GE eset generation
